src/Database/SqlDialectTrait.php

```diff
@@ -178,6 +178,34 @@ protected function _deleteQueryTranslator($query)
         if (!$hadAlias) {
             return $query;
         }
+
+        return $this->_removeAliasesFromConditions($query);
+    }
+
+    /**
+     * Apply translation steps to update queries.
+     *
+     * Chops out aliases on update query conditions as not all database dialects do support
+     * aliases in update queries.
+     *
+     * Just like for delete queries, joins are currently not supported for update queries.
+     *
+     * @param \Cake\Database\Query $query The query to translate
+     * @return \Cake\Database\Query The modified query
+     */
+    protected function _updateQueryTranslator($query)
+    {
+        return $this->_removeAliasesFromConditions($query);
+    }
+
+    /**
+     * Removes aliases from the `WHERE` clause of a query.
+     *
+     * @param \Cake\Database\Query $query The query to process.
+     * @return \Cake\Database\Query The modified query.
+     */
+    protected function _removeAliasesFromConditions($query)
```

Comment:
Ah OK, I see... a quick test shows that it doesn't, which I guess is because unline `QueryExpression::traverse()`, `Query::traverse()` isn't recursive, so for queries it will just pass the parts to the visotor, which will discard them at the `instanceof Comparison` check.

I'll have a closer look later on and add some subqueries to the test.